aws/resource_aws_kinesis_stream.go

```diff
@@ -85,36 +92,54 @@ func resourceAwsKinesisStreamCreate(d *schema.ResourceData, meta interface{}) er
 		StreamName: aws.String(sn),
 	}
 
-	_, err := conn.CreateStream(createOpts)
-	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			return fmt.Errorf("[WARN] Error creating Kinesis Stream: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
-		}
-		return err
-	}
+	attemptCount := 1
+	for attemptCount <= KINESIS_MAX_ATTEMPTS {
```

Comment:
Instead of (re)building a complex retry logic, counting attempts and time elapsed I think we should use the helper from [`helper/resource`](https://github.com/hashicorp/terraform/blob/master/helper/resource/wait.go#L10)

i.e. 
```go
err := resource.Retry(30 * time.Second, func() *resource.RetryableError {
    _, err := conn.CreateStream(createOpts)
    if isAWSErr(err, "LimitExceededException", "simultaneously be in CREATING or DELETING") {
        return resource.RetryableError(err)
    }
    return resource.NonRetryableError(err)
})
```

so we could reduce ~20 LOC to (IMO) more readable ~7 lines.